source/opt/add_var_for_funccall_param.h

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef _VAR_FUNC_CALL_PASS_H
+#define _VAR_FUNC_CALL_PASS_H
+
+#include "source/opt/pass.h"
+
+namespace spvtools {
+namespace opt {
+class AddVarsForFuncCallParamPass : public Pass {
+ public:
+  AddVarsForFuncCallParamPass() {}
+  const char* name() const override { return "add-var-for-funcall-param"; }
+  Status Process() override;
+  IRContext::Analysis GetPreservedAnalyses() override {
+    return IRContext::kAnalysisDefUse | IRContext::kAnalysisDecorations |
+           IRContext::kAnalysisInstrToBlockMapping;
```

Comment:
Won't this keep more analyses alive than this?  It should not have to invalidate the CFG, right?  You can see the full list [here](https://github.com/KhronosGroup/SPIRV-Tools/blob/2b2b0282af25779a5ae5dfc08a1c4e3be5a8ae17/source/opt/ir_context.h#L64).  I don't think any of them need to be invalidated.